ipapython/ipautil.py

```diff
@@ -1571,3 +1572,107 @@ def major(self):
     @property
     def minor(self):
         return self[1]
+
+
+class _TemporaryDirectoryManager:
+    """File manager for shared temporary files
+
+    The temporary directory manager simplifies the handling of temporary
+    files that are shared with other processes or kept through out the life
+    time of the current process. It should only be used in case
+    tempfile.NamedTemporaryFile is not up for the task.
```

Comment:
I think it would be helpful for future developers to expand on a few specific cases where tempfile is not up to the task.